src/lib/Libifl/pbs_loadconf.c

```diff
@@ -257,6 +257,39 @@ parse_config_line(FILE *fp, char **key, char **val)
 	return ret;
 }
 
+/**
+ * @brief frame the psi_t struct from svr instance id passed
+ * 
+ * @param[in,out] psi - server instance id
+ * @param[in] svr_id - server instance id
+ * @return int 
+ * @retval 0 - success
+ * @retval !0 - failure
+ */
+int
+frame_psi(psi_t *psi, char *svr_id)
+{
+	char *svrname;
+
+	if (!psi || !svr_id)
+		return -1;
+
+	svrname = parse_servername(svr_id, &psi->port);
+	if (svrname == NULL)
+		return -1;
+
+	strcpy(psi->name, svrname);
```

Comment:
Could result in a buffer overflow, since the size of psi->name is smaller than what svrname can return